heartbeat/monitors/logger/logger.go

```diff
@@ -54,46 +63,65 @@ func SetLogger(l *logp.Logger) *logp.Logger {
 }
 
 func getLogger() *logp.Logger {
+	mtx.Lock()
+	defer mtx.Unlock()
+
 	if eventLogger == nil {
 		return SetLogger(logp.L())
 	}
 
 	return eventLogger
 }
 
-func extractRunInfo(event *beat.Event) (*monitorRunInfo, error) {
-	monitorID, mIDErr := event.GetValue("monitor.id")
-	durationUs, dErr := event.GetValue("monitor.duration.us")
-	monType, tErr := event.GetValue("monitor.type")
-	if mIDErr != nil || dErr != nil || tErr != nil {
-		getLogger().Errorw(
-			"Error gathering information to log event",
-			logp.Errors("logErrors", []error{mIDErr, dErr, tErr}),
-		)
-
-		return nil, errors.New("error gathering information to log event")
+func extractRunInfo(event *beat.Event) (*MonitorRunInfo, error) {
+	errors := []error{}
+	monitorID, err := event.GetValue("monitor.id")
+	if err != nil {
+		errors = append(errors, err)
 	}
 
-	durationMs := time.Duration(durationUs.(int64) * int64(time.Microsecond)).Milliseconds()
-	runInfo := NewMonitorRunInfo(
-		monitorID.(string),
-		monType.(string),
-		durationMs,
-	)
+	durationUs, err := event.GetValue("monitor.duration.us")
+	if err != nil {
+		errors = append(errors, err)
+	}
 
-	return &runInfo, nil
+	monType, err := event.GetValue("monitor.type")
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	status, err := event.GetValue("monitor.status")
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		return nil, fmt.Errorf("logErrors: %+v", errors)
```

Comment:
This pattern is 100% fine. Another one that works is to just keep re-wrapping errors using `fmt.Errorf("something %w", err)`. That said, it just doesn't matter for this use case since we'll never use `errors.Is` and friends. I just thought I'd mention it, because it's neat. It's a little weird though, because it breaks the hierarchical metaphor of go errors, and takes them further from the more mainstream way exceptions work.